goal_src/pc/pckernel.gc

```diff
@@ -760,15 +805,28 @@
 
 (defmethod new pc-settings ((allocation symbol) (type-to-make type))
   "make a new pc-settings"
-
   (let ((obj (object-new allocation type-to-make (the-as int (-> type-to-make size)))))
-
     (reset obj)
-
-    obj
-    )
-  )
-
+    ;; auto load settings if available
+    ;; if saved settings are corrupted, reset them
+    (clear *pc-temp-string-1*)
+    (format *pc-temp-string-1* "~S/pc-settings.gc" *pc-settings-folder*)
+
+    (if (pc-filepath-exists? *pc-temp-string-1*)
+      (begin
+        (format 0 "[PC] PC Settings found at '~S'...loading!~%" *pc-temp-string-1*)
+        (if (read-from-file obj *pc-temp-string-1*)
+          (return obj)
+          (begin
+            (format 0 "[PC] PC Settings found at '~S' but could not be loaded, re-creating with defaults!~%" *pc-temp-string-1*)
+            (reset obj))))
+      (format 0 "[PC] PC Settings not found at '~S'...creating and initializing with defaults!~%" *pc-temp-string-1*))
+
+    ;; persist reset / new settings if we made it this far
+    (pc-mkdir-path *pc-temp-string-1*)
+    (commit-to-file obj)
+    (clear *pc-temp-string-1*)
```

Comment:
And this - is it just writing the settings after they've been reset? Probably not super useful (reading the default file and not reading a file at all is the exact same thing), and personally I'd rather it not write things at start-up in case things go wrong.